packages/core-v2/src/visual-model/default-visual-model.spec.ts

```diff
@@ -32,20 +32,55 @@ test("Set and delete model color.", () => {
 
 test("Set and delete visual entity.", () => {
   const model = createModel("abc");
+  const numberOfEntitiesInModelAfterInitialization = [...model.getVisualEntities().entries()].length;
   //
-  expect(model.getVisualEntityForRepresented("s")).toBeNull();
+  expect(model.getVisualEntitiesForRepresented("s").length).toBe(0);
   model.addVisualNode({
     representedEntity: "s",
     model: "m",
     content: [],
     visualModels: [],
     position: { x: 100, y: 200, anchored: null },
   });
-  const actual = model.getVisualEntityForRepresented("s");
-  expect(actual).not.toBeNull();
-  expect(isVisualNode(actual as VisualEntity)).toBeTruthy();
-  const visualNode = actual as VisualNode;
+  const actual = model.getVisualEntitiesForRepresented("s");
+  expect(actual.length).toBeGreaterThan(0);
+  expect(isVisualNode(actual[0] as VisualEntity)).toBeTruthy();
+  const visualNode = actual[0] as VisualNode;
   expect(visualNode.representedEntity).toBe("s");
   model.deleteVisualEntity(visualNode.identifier);
-  expect(model.getVisualEntityForRepresented("s")).toBeNull();
+  expect(model.getVisualEntitiesForRepresented("s").length).toBe(0);
+  expect([...model.getVisualEntities().entries()].length).toBe(numberOfEntitiesInModelAfterInitialization);
+});
+
+test("Set and delete visual entity (multi).", () => {
+  const model = createModel("abc");
+  //
+  expect(model.getVisualEntitiesForRepresented("s").length).toBe(0);
+  model.addVisualNode({
+    representedEntity: "s",
+    model: "m",
+    content: [],
+    visualModels: [],
+    position: { x: 100, y: 200, anchored: null },
+  });
+  model.addVisualNode({
+    representedEntity: "s",
+    model: "m",
+    content: [],
+    visualModels: [],
+    position: { x: 200, y: 300, anchored: null },
+  });
+  const actual = model.getVisualEntitiesForRepresented("s");
+  expect(actual.length).toBe(2);
```

Comment:
After this line you could use destrcutive assignment and grab the first and the second item.
Then there is no neet to repeatedly write `actual[0]` and `actual[1]`.